sources/packages/web/src/components/aest/students/modals/UpdateDisabilityStatusModal.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <v-form ref="updateDisabilityStatusForm">
+    <modal-dialog-base
+      title="Update disability status"
+      :showDialog="showDialog"
+    >
+      <template #content>
+        <error-summary :errors="updateDisabilityStatusForm.errors" />
+        <p class="label-value">
+          Update the disability status of student and add note.
+        </p>
+        <v-select
+          hide-details="auto"
+          label="Disability Status"
+          density="compact"
+          :items="disabilityStatusSelectItems"
+          v-model="formModel.disabilityStatus"
+          variant="outlined"
+          :rules="[(v) => checkNullOrEmptyRule(v, 'Disability Status')]"
+        />
+        <v-textarea
+          v-model="formModel.noteDescription"
+          variant="outlined"
+          label="Note"
+          :rules="[checkNotesLengthRule]"
+          required
+          class="mt-4"
+          hide-details="auto"
+        ></v-textarea>
+      </template>
+      <template #footer>
+        <check-permission-role :role="allowedRole">
+          <template #="{ notAllowed }">
+            <footer-buttons
+              primaryLabel="Update"
+              :processing="loading"
+              @secondaryClick="cancel"
+              @primaryClick="submit"
+              :disablePrimaryButton="notAllowed"
+            />
+          </template>
+        </check-permission-role>
+      </template>
+    </modal-dialog-base>
+  </v-form>
+</template>
+<script lang="ts">
+import { PropType, ref, reactive, defineComponent } from "vue";
+import { useModalDialog, useRules, useFormatters } from "@/composables";
+import { Role, VForm, SelectItemType, DisabilityStatus } from "@/types";
+import { UpdateDisabilityStatusAPIInDTO } from "@/services/http/dto";
+import ErrorSummary from "@/components/generic/ErrorSummary.vue";
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
```

Comment:
Suggestion: these 2 components are used in many places, may be we can import it to `main.ts`